### PR TITLE
Added plot for predictive probabilities over time

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Added plot for predictive probabilities over time.

Adressing issue #44

### CODE REVIEW

File: src/psycopt2d/visualization/prob_over_time.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    plot_df["patient_id"] = plot_df["patient_id"].astype(str)
+    plot_df["label"] = plot_df["label"].astype(str)
+
+    max_pred_prob = plot_df.groupby(["patient_id"])["pred_prob"].max()
+    plot_df["color"] = [max_pred_prob[id_] for id_ in plot_df["patient_id"]]
+
+    chart = (
+        alt.Chart(plot_df)
+        .mark_line(opacity=0.3)
```
Comment (martbern) [inline src/psycopt2d/visualization/prob_over_time.py:86]: Might wanna add opacity as a function argument? :-)

We want to be able to set lower opacities for plots with a higher number of lines.

**Review by martbern — APPROVED** [review]
Excellent work, Kenneth! Looking forward to applying it on real world data.

Very minor comments below, then feel free to merge 👍

File: src/psycopt2d/visualization/prob_over_time.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+def plot_prob_over_time(
+    timestamp: Iterable[datetime],
+    pred_prob: Iterable[float],
+    label: Iterable[Union[int, str]],
+    outcome_timestamp: Iterable[Union[datetime, None]],
+    patient_id: Iterable[Union[int, str]],
+    x_axis: str = "Time from outcome",
+    y_axis: str = "Model Predictive Probability",
+    legend: str = "Highest Predictive Probability",
+    look_behind: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply src/psycopt2d/visualization/prob_over_time.py:20]: `shade_lookbehind_distance` indicates it is a boolean variable, which it isn't.

File: src/psycopt2d/visualization/prob_over_time.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            positive outcome.
+        patient_id (Iterable[Union[int, str]]): patient ID. Used for plotting one line
+            pr. patient.
+        x_axis (str, optional): Label on x-axis. Defaults to "Time from outcome".
+        y_axis (str, optional): Label of y-axis. Defaults to "Model Predictive
+            Probability".
+        legend (str, optional): Label on legend. Defaults to "Highest Predictive
+            Probability".
+        look_behind (Optional[int], optional): Look-behind window. Defaults to None in
+            which case no shaded areas is plotted.
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply src/psycopt2d/visualization/prob_over_time.py:38]: please don't make so large suggestions. Make correction hard and essentially these are 6 independent changes.

**Pull request merged**